multicluster/controllers/multicluster/leader/stale_controller.go

```diff
@@ -0,0 +1,273 @@
+	// failed to clean up all stale ResourceExports.
+	cleanUpRetry = wait.Backoff{
+		Steps:    5,
+		Duration: 500 * time.Millisecond,
+		Factor:   1.0,
+		Jitter:   1,
+	}
+)
+
+// LeaderStaleResCleanupController will run periodically (memberClusterAnnounceStaleTime / 2 ) (12 hours)
+// to clean up stale MemberClusterAnnounce resources in the leader cluster if the MemberClusterAnnounce
+// timestamp annotation has not been updated for memberClusterAnnounceStaleTime (24 hours).
+// It will remove all ResourceExports belong to a member cluster when the corresponding MemberClusterAnnounce
+// CR is deleted. It will also try to clean up all stale ResourceExports during start.
+type LeaderStaleResCleanupController struct {
+	client.Client
+	Scheme    *runtime.Scheme
+	namespace string
+}
+
+func NewLeaderStaleResCleanupController(
+	Client client.Client,
+	Scheme *runtime.Scheme,
+	namespace string,
+) *LeaderStaleResCleanupController {
+	reconciler := &LeaderStaleResCleanupController{
+		Client:    Client,
+		Scheme:    Scheme,
+		namespace: namespace,
+	}
+	return reconciler
+}
+
+func cleanUpAllMemberClusterAnnounces(ctx context.Context, mgrClient client.Client) error {
+	memberClusterAnnounceList := &mcv1alpha1.MemberClusterAnnounceList{}
+	if err := mgrClient.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		klog.ErrorS(err, "Fail to get MemberClusterAnnounce")
+		return err
+	}
+	for _, mca := range memberClusterAnnounceList.Items {
+		if err := mgrClient.Delete(ctx, &mca, &client.DeleteOptions{}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// cleanUpStaleMemberClusterAnnounces will delete any MemberClusterAnnounce if its
+// last update timestamp is over 24 hours.
+func (c *LeaderStaleResCleanupController) cleanUpStaleMemberClusterAnnounces(ctx context.Context) {
+	memberClusterAnnounceList := &mcv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{Namespace: c.namespace}); err != nil {
+		klog.ErrorS(err, "Fail to get MemberClusterAnnounce in the Namespace", "namespace", c.namespace)
+		return
+	}
+
+	for _, m := range memberClusterAnnounceList.Items {
+		memberClusterAnnounce := m
+		lastUpdateTime, err := time.Parse(time.RFC3339, memberClusterAnnounce.Annotations[commonarea.TimestampAnnotationKey])
+		if err == nil && time.Since(lastUpdateTime) < memberClusterAnnounceStaleTime {
+			continue
+		}
+		if err == nil {
+			klog.InfoS("Cleaning up stale MemberClusterAnnounce. It has not been updated within the agreed period", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce), "agreedPeriod", memberClusterAnnounceStaleTime)
+		} else {
+			klog.InfoS("Cleaning up stale MemberClusterAnnounce. The latest update time is not in RFC3339 format", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
+		}
+
+		if err := c.Client.Delete(ctx, &memberClusterAnnounce, &client.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
When the member controller is updating the MemberClusterAnnounce, it will check the existing MemberClusterAnnounce before updating every time. If it's deleted, it will create a new one.
The stale controller in the leader will delete those stale MemberClusterAnnounce over 2 hours, if a ClusterSet in a member is disconnected from the leader over 2 hours, the status of member's ClusterSet readiness will be false. When the member controller create a new memberClusterAnnounce successfully and update the  ClusterSet readiness to true. The member controllers will re-sync all exported resources in this PR https://github.com/antrea-io/antrea/pull/5410/